src/ConnectionFactory.php

```diff
@@ -75,4 +83,31 @@ protected function getDefaultConfig($configKey)
             'doctrine_commented_types' => [],
         ];
     }
+
+    /**
+     * Registers all declared typed, if not already done.
+     *
+     * @param ContainerInterface $container
+     */
+    private function registerTypes(ContainerInterface $container)
+    {
+        if (self::$areTypesRegistered) {
+            return;
+        }
+
+        $applicationConfig = $container->has('config') ? $container->get('config') : [];
```

Comment:
I can't this is desired behavior and tested I guess because this test fails https://github.com/DASPRiD/container-interop-doctrine/blob/master/test/ConnectionFactoryTest.php#L40 but I agree with you @Ocramius this should fail, but I guess this needs to be changed in more places so maybe leave it for next time or remove the test case :smile: